### PR TITLE
Produce blocks every 1 second for devnet

### PR DESCRIPTION
**Pull request by bowenwang1996**
In devnet, instead of producing blocks for every transaction, we now have the option (in serviceConfig) to produce blocks every 1 second to be more efficient.

### CODE REVIEW

File: node/cli/src/lib.rs [hunk 1]
```diff
@@ -113,6 +111,13 @@ pub fn get_service_config() -> service::ServiceConfig {
               .help("Sets public key to sign with, \
                          can be omitted with 1 file in keystore")
               .takes_value(true)
+        ).arg(
+            Arg::with_name("prod_block_on_tx")
```
Comment (ilblackdragon) [inline node/cli/src/lib.rs]: --immediate

Comment (ilblackdragon) [reply node/cli/src/lib.rs]: also can revert, and make it --batch_transactions to batch them and by default it's immediate.

File: node/cli/src/service.rs [hunk 2]
```diff
@@ -139,6 +137,8 @@ pub struct ServiceConfig {
     pub chain_spec_path: Option<PathBuf>,
     pub log_level: log::LevelFilter,
     pub rpc_port: u16,
+    /// whether to produce blocks with a set time period
+    pub batch_transactions: bool,
```
Comment (nearmax) [inline node/cli/src/service.rs]: I don't think it is a good idea to mix DevNet-specific and general configs. This needs to be separated. I suggest we always batch transactions for the DevNet, which means we do not need this flag.

Comment (bowenwang1996) [reply node/cli/src/service.rs]: Tests will fail if we batch transactions

File: node/consensus/src/passthrough.rs [hunk 3]
```diff
@@ -15,11 +20,37 @@ pub fn spawn_consensus(
...
+                tokio::spawn(consensus_tx.clone().send(c).map(|_| ()).map_err(|e| {
+                    error!("Failure sending pass-through consensus {:?}", e);
+                }));
+                future::ok((consensus_tx, control_rx))
+            })
+            .map(|_| ())
+            .map_err(|_| ());
+        tokio::spawn(task);
+    } else {
+        let message_queue = Arc::new(Mutex::new(vec![]));
```
Comment (nearmax) [inline node/consensus/src/passthrough.rs]: We can avoid creating a separate queue with a mutex by merging two streams: Interval and payload_rx. Using `merge` or `select` method of streams, then applying the same fold to batch payload together.

File: node/cli/src/service.rs (hunk 2 above)
Comment (nearmax) [reply node/cli/src/service.rs]: The current mode of running the DevNet is with batched transactions. If we add a boolean command flag that is only used to trigger a different execution path for tests then the tests will not be testing the real execution path. In general, adding a boolean flag just to hack around the tests is not okay.

**Comment by ilblackdragon** [comment]
@bowenwang1996 https://github.com/nearprotocol/nearcore/pull/373 adds re-try to poll for results in integration tests instead of waiting, so should not be an issue now and not require `immediate` mode.

**Pull request merged**